frontend_lib/src/component/Tags/NewTagForm.jsx

```diff
@@ -36,6 +36,21 @@ export class NewTagForm extends React.Component {
     this.setState({ tagName: '', autoCompleteActive: false })
   }
 
+  componentDidMount () {
+    document.addEventListener('click', this.handleClickEvent, true)
+  }
+
+  componentWillUnmount () {
+    document.removeEventListener('click', this.handleClickEvent, true)
+  }
+
+  handleClickEvent = (event) => {
+    const { state } = this
+    const inputField = document.getElementById('addTagFullInput')
+
+    this.setState({ autoCompleteActive: inputField.contains(event.target) && state.tagName })
```

Comment:
as seen together, try using react-onclickoutside and add explicit test for state.tagName